u_parrots.go

```diff
@@ -17,26 +17,18 @@ import (
 )
 
 var ErrUnknownClientHelloID = errors.New("tls: unknown ClientHelloID")
-var ErrNotPSKClientHelloID = errors.New("tls: ClientHello does not contain pre_shared_key extension")
-var ErrPSKExtensionExpected = errors.New("tls: pre_shared_key extension expected when fetching preset ClientHelloSpec")
 
 // UTLSIdToSpec converts a ClientHelloID to a corresponding ClientHelloSpec.
 //
 // Exported internal function utlsIdToSpec per request.
-func UTLSIdToSpec(id ClientHelloID, pskExtension ...PreSharedKeyExtension) (ClientHelloSpec, error) {
-	if len(pskExtension) > 1 {
-		return ClientHelloSpec{}, errors.New("tls: at most one PreSharedKeyExtensions is allowed")
-	}
-
-	chs, err := utlsIdToSpec(id)
-	if err != nil && errors.Is(err, ErrUnknownClientHelloID) {
-		chs, err = utlsIdToSpecWithPSK(id, pskExtension...)
-	}
-
-	return chs, err
+func UTLSIdToSpec(id ClientHelloID, pskExt PreSharedKeyExtension, sessionTicketExt *SessionTicketExtension) (ClientHelloSpec, error) {
```

Comment:
Since this is a breaking change, maybe consider `//deprecated` the old interface and update it to use preset/default parameters, while introducing a new interface. Otherwise it somehow breaks existing code and may require a major version change (v1 -> v2).

Reply:
I felt that by default we may want to provide an interface without requiring ANY of these extensions and use default behaviors (random session id, psk, etc). And based on which we could introduce new interfaces to OPTIONALLY set `SessionTicketExtension` or `PreSharedKeyExtension`.

Reply:
I have reverted the changes to existing public interfaces.

Two new public methods are introduced 
> Allow session tickets to be set through `SetSessionTicketExtension` and PSK to be set through `SetPskExtension`.

Reply:
Looks good!